packages/kbn-unified-data-table/src/components/data_table_document_selection.tsx

```diff
@@ -71,13 +75,13 @@ export const SelectButton = (props: EuiDataGridCellValueElementProps) => {
   );
 };
 
-export const SelectAllButton = () => {
-  const { selectedDocsState, pageIndex, pageSize, rows } = useContext(UnifiedDataTableContext);
+export const getSelectAllButton = (rows: DataTableRecord[]) => () => {
+  const { selectedDocsState, pageIndex, pageSize } = useContext(UnifiedDataTableContext);
   const { getCountOfFilteredSelectedDocs, deselectSomeDocs, selectMoreDocs } = selectedDocsState;
 
   const docIdsFromCurrentPage = useMemo(() => {
     return getDocIdsForCurrentPage(rows, pageIndex, pageSize);
-  }, [rows, pageIndex, pageSize]);
+  }, [pageIndex, pageSize]);
```

Comment:
Yes because `getSelectAllButton` is now a function that accepts `rows` as a param and returns a component, so `rows` in no longer a part of the React lifecycle and will only change whenever `getSelectAllButton` is called.